jerry-core/mem/mem-poolman.c

```diff
@@ -34,16 +34,21 @@
 #define MEM_ALLOCATOR_INTERNAL
 #include "mem-allocator-internal.h"
 
+ /**
+ * Type for mem_pool_chunk
+ */
+typedef struct mem_pool_chunk mem_pools_chunk_t;
+
 /**
  * Node for free chunk list
  */
-typedef struct
+struct mem_pool_chunk
 {
-  struct mem_pools_chunk_t *next_p; /* pointer to next pool chunk */
+  mem_pools_chunk_t *next_p; /* pointer to next pool chunk */
 #ifndef MEM_HEAP_PTR_64
   uint32_t dummy; /* dummy member for alignment */
 #endif
-} mem_pools_chunk_t;
+} mem_pool_chunk;
```

Comment:
This rewrite doesn't seem to be equivalent to the original version. This code defines a `struct mem_pool_chunk` -- and then also defines a variable named `mem_pool_chunk` of type `struct mem_pool_chunk`. This global variable is certainly not what we want.

BTW, why do we need this rewrite?